dll.py

```diff
@@ -1,52 +1,163 @@
-        self._prev = prev
-        self._next = next
+        self.prev = prev
+        self.next = next
+
+    @property
+    def value(self) -> "Vertex":
+        return self._vertex
+
+    def __str__(self):
+        return str(self._vertex)
 
 
 class DoubleLinkedList(object):
-    def __init__(self, node: "Vertex"=None):
-        self._first_node = node
-        self._last_node = node
-
-    def insert_after(self, node, new_node):
-        new_node.prev = node
-        if node.next is None:
-            new_node.next = None
-            self._last_node = new_node
-        else:
-            new_node.next = node.next
-            node.next.prev = new_node
-        node.next = new_node
-
-    def insert_before(self, node, new_node):
-        new_node.next = node
-        if node.prev is None:
-            new_node.prev = None
-            self._first_node = new_node
+    def __init__(self, val: "Vertex" = None):
+        self._head = None
+        self._tail = None
+        self._size = 0
+        if val:
+            self.pushleft(val)
+
+    def pushleft(self, val: "Vertex"):
+        """Add element to the start of the list"""
+        new_node = Node(val)
+        if self._size == 0:
+            self._head = new_node
+            self._tail = new_node
         else:
-            new_node.prev = node.prev
-            node.prev.next = new_node
-        node.prev = new_node
-
-    def append(self, node):
-        if self._first_node is None:
-            self._first_node = node
-            self._last_node = node
-            node.prev = None
-            node.next = None
+            self._head.prev = new_node
+            new_node.next = self._head
+            self._head = new_node
+        self._size += 1
+
+    def push(self, val: "Vertex"):
+        """Add element to the end of the list"""
+        new_node = Node(val)
+        if self._size == 0:
+            self._head = new_node
+            self._tail = new_node
         else:
-            self.insert_after(self.last_node, node)
+            self._tail.next = new_node
+            new_node.prev = self._tail
+            self._tail = new_node
+        self._size += 1
+
+    # def insert_after(self, node, val: "Vertex"):
+    #     new_node = Node(val)
+    #     new_node.prev = node
+    #     if node.next is None:
+    #         new_node.next = None
+    #         self._tail = new_node
+    #     else:
+    #         new_node.next = node.next
+    #         node.next.prev = new_node
+    #     node.next = new_node
+    #     self._size += 1
+    #
+    # def insert_before(self, node, val: "Vertex"):
+    #     new_node = Node(val)
+    #     new_node.next = node
+    #     if node.prev is None:
+    #         new_node.prev = None
+    #         self._head = new_node
+    #     else:
+    #         new_node.prev = node.prev
+    #         node.prev.next = new_node
+    #     node.prev = new_node
+    #     self._size += 1
+    #
+    # def append(self, node):
```

Comment:
We kunnen hier dus voor 'append' en 'appendleft' gaan of 'push' en 'pushleft'. Standaard is blijkbaar vanaf de achterkant dingen toevoegen/weghalen.